torch/distributed/_shard/checkpoint/state_dict_saver.py

```diff
@@ -57,7 +62,7 @@ def _prepare(
     storage_key_to_fqn: Dict[str, str] = dict()
     # The assumption is that all non ShardedTensor items are replicated
     #   and we can save them from rank 0.
-    write_replicated_data = not (dist.is_initialized() and dist.get_rank() != 0)
+    write_replicated_data = not (dist.is_initialized() and dist.get_rank(process_group) != 0)
```

Comment:
same as above, just enforce `dist.is_initialized()` and throw an error otherwise.